temporalio/worker/_workflow_instance.py

```diff
@@ -1774,6 +1774,8 @@ def _convert_payloads(
             # Don't wrap payload conversion errors that would fail the workflow
             raise
         except Exception as err:
+            if self._is_workflow_failure_exception(err):
```

Comment:
_Technically_ this has backwards compatibility concerns since it changes existing behavior based on raised exception, but I think it's ok in this instance.

But can we change the PR title to be a bit more generic so when it shows up in release notes it doesn't look like it's Pydantic specific? Maybe something like "do not wrap workflow-failure exceptions from converters in workflows"?